pkg/api/start.go

```diff
@@ -60,19 +55,40 @@ func Setup(assets *embed.FS) (*chi.Mux, error) {
 	r.Use(middleware.Logger)
 	r.Use(middleware.Recoverer)
 
-	ctx := context.Background()
-	cache, err := resources.NewCache(ctx)
+	ctx, cancel := context.WithCancel(context.Background())
+	k8sClient, err := k8s.NewClient()
 	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to create k8s client: %w", err)
+	}
+
+	cache, err := resources.NewCache(ctx, k8sClient)
+	if err != nil {
+		cancel()
 		return nil, fmt.Errorf("failed to create cache: %w", err)
 	}
 
+	// Create the disconnected channel
+	disconnected := make(chan error)
```

Comment:
```suggestion
	disconnected := make(chan error)
	defer close(disconnected)
```

Don't think the channel is being closed

Reply:
looking at this some more and chatting with the machines, i don't think we need to close this channel. the channel needs to stay open for the lifetime of the app, and if the app exits for whatever reason then the channel will be closed anyway. If there's another reason for closing it, then we'll need to close it differently than `defer` so it doesn't close when `Setup()` returns.

Reply:
@TristanHoladay tested locally, closing channel here breaks connection flow